packages/flutter/lib/src/painting/continuous_stadium_border.dart

```diff
@@ -0,0 +1,348 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'basic_types.dart';
+import 'borders.dart';
+import 'edge_insets.dart';
+
+/// A stadium border with continuous corners.
+///
+/// A shape similar to a stadium, but with a smoother transition from
+/// each linear edge to its 180º curves. Each 180º curve is approximately half
+/// an ellipse.
+///
+/// In this shape, the curvature of each 180º curve around the length of the arc
+/// is approximately a gaussian curve instead of a step function as with a
+/// traditional half circle round.
+///
+/// In an attempt to keep the shape of the rectangle the same regardless of its
+/// dimension (and to avoid clipping of the shape), the radius will
+/// automatically be lessened if its width or height is less than ~3x the
+/// declared radius. The new resulting radius will always be maximal in respect
+/// to the dimensions of the given rectangle.
+///
+/// The ~3 represents twice the ratio (ie. ~3/2) of a corner's declared radius
+/// and the actual height and width of pixels that are manipulated to render it.
+/// For example, if a rectangle had dimensions 80px x 100px, and a corner radius
+/// of 25, in reality ~38 pixels in each dimension would be used to render a
+/// corner and so ~76px x ~38px would be used to render both corners on a given
+/// side.
+///
+/// The two 180º arcs will always be positioned on the shorter side of the
+/// rectangle like with the traditional [StadiumBorder] shape.
+///
+/// {@tool sample}
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Material(
+///     color: Colors.blueAccent[400],
+///     shape: const ContinuousStadiumBorder(),
+///     child: const SizedBox(
+///       height: 100.0,
+///       width: 200.0,
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// The image below depicts a [ContinuousStadiumBorder] with a width of 200
+/// and a height of 100.
+///
+/// ![](https://flutter.github.io/assets-for-api-docs/assets/painting/Shape.continuous_stadium.png)
+///
+/// See also:
+///
+/// * [RoundedRectangleBorder], which is a rectangle whose corners are
+///   precisely quarter circles.
+/// * [ContinuousRectangleBorder], which is a rectangle whose 4 edges have
+///   a continuous transition into each of its four corners.
+/// * [StadiumBorder], which is a rectangle with semi-circles on two parallel
+///   edges.
+class ContinuousStadiumBorder extends ShapeBorder {
+  /// Creates a continuous stadium border.
+  ///
+  /// The [side], argument must not be null.
+  const ContinuousStadiumBorder({
+    this.side = BorderSide.none,
+  }) : assert(side != null);
+
+  /// The style of this border.
+  ///
+  /// If the border side width is larger than 1/10 the length of the smallest
```

Comment:
Some question here for the 1/10

Reply:
Why is 1/10 relevant here? The overlapping shapes are never concentric no? Just say that.